client/components/domains/register-domain-step/index.jsx

```diff
@@ -1406,6 +1424,9 @@ class RegisterDomainStep extends Component {
 	onSearch = async ( searchQuery ) => {
 		debug( 'onSearch handler was triggered with query', searchQuery );
 
+		// Trigger a clean state when a new search is starting
+		this.onSearchChange( searchQuery );
```

Comment:
I think this is fine, but I wonder if calling `onSearchChange` every time here would have unintended consequences 🤔 
My suggestion would be removing this call and adding these properties to the `setState` call right below:
```js
				// Clean errors from previous searches
				showAvailabilityNotice: false,
				availabilityError: null,
				availabilityErrorData: null,
				showSuggestionNotice: false,
				suggestionError: null,
				suggestionErrorData: null,
				suggestionErrorDomain: null,
```
That would only clean previous searches' errors.